part-3/part-3.md

```diff
@@ -152,252 +153,289 @@ const source = from([1, 2, 3]).pipe(
+**withLatestFrom()** combines the source `Observable` with another one, and emits a combined value only when the source one emits.
 
 ```ts
-import { timer } from "rxjs";
+import { interval } from 'rxjs';
 
-import { withLatestFrom } from "rxjs/operators";
+import { withLatestFrom } from 'rxjs/operators';
 
-// Первое значение вернет сразу, а остальные каждую секунду
+const sourceFirst = interval(1000);
 
-const sourceFirst = timer(0, 1000);
-
-// Первое значение через 0.5 секунды, а остальные каждую секунду
-
-const sourceSecond = timer(500, 1000);
+const sourceSecond = interval(500);
 
 const source = sourceFirst.pipe(withLatestFrom(sourceSecond));
 
-// Выведет в консоль:
+// Will log:
 
-//. [1, 0] через 1.0 sec
+//. [0, 1] in 1.0 second
 
-// [2, 1] через 2.0 sec
+// [1, 3] in 2.0 seconds
 
-// [3, 2] через 3.0 sec
+// [2, 5] in 3.0 seconds
 
-source.subscribe((value) => console.log(value));
+source.subscribe(value => console.log(value));
 ```
 
-Разница с предыдущим примером в том, что теперь новое значение создается только тогда, когда изменяется основной поток.
+The only difference from **combineLatest** is that it will only emit when the source emits. But it won't emit if the other `Observable` has not emitted at all, even if the source emits, just like **combineLatest**. 
 
 #
 
-# **Обработка ошибок в потоках**
+# **Handling errors in streams**
 
-В реальных приложениях могут происходить исключительные ситуации, и нам необходимо их правильно обработать. Давайте создадим исключение в потоке, чтобы посмотреть что произойдет.
+In real life application some exceptional situations may arise, in which cases we will need to correctly handle errors. Let's create an error and see how it goes.
 
 ```ts
-import { from } from "rxjs";
+import { from } from 'rxjs';
 
-import { map } from "rxjs/operators";
+import { map } from 'rxjs/operators';
 
 const source = from([1, 2, 3]).pipe(
-  // Что-то пошло не так
+  // Something went wrong!
 
   map(() => {
-    thrownewError("Unexpected 🙀!");
+    throw newError("Unexpected 🙀!");
   })
 );
 
-// Выбросится исключение:
+// An Error will be thrown:
 
 //. Error: Unexpected 🙀!
 
-source.subscribe((value) => console.log(value));
+source.subscribe(value => console.log(value));
 ```
 
-И в данном случае это исключение никак не обрабатывается. Чтобы его обработать, необходимо воспользоваться оператором **catchError()**.
+In this case, this `Error` is not being handled at all.To handle it, we will need o learn how to use the **catchError()** operator.
 
-**catchError()** обрабатывает все исключения, которые происходят в потоке. Обратите внимание, что старый поток завершается, поэтому оператор возвращает новый поток.
+**catchError()** handles all the errors that happen inside the stream. Notice that when an error happens, the old stream completes, so we need to return a new stream from that operator.
```

Comment:
```suggestion
**catchError()** handles all errors that occur inside of a stream. Notice that when an error occurs, the old stream completes, so we need to return a new stream from the `catchError` operator.
```

Reply:
Consider mentioning what happens if we don't return an observable from `catchError`.